app.rb

```diff
@@ -41,12 +41,22 @@ def create_a_person
     puts 'Name'
     person_name = gets.chomp
 
-    case is_student
+    create_person_value(is_student, age, person_name)
+  end
+
+  def create_person_value(person, age, person_name)
+    case person
     when 1
       puts 'Has parent permission? [Y/N]'
-      gets.chomp.downcase
-      student = Student.new(age: age, name: person_name)
-      @persons.push(student)
+      response = gets.chomp.downcase
+      if response == 'n'
+        puts 'Access denied'
+      elsif response == 'y'
+        student = Student.new(age: age, name: person_name)
+        @persons.push(student)
+      else
+        puts 'Out of range'
+      end
```

Comment:
**App.rb:**

- **Single-Responsibility Principle:**
  - Refactor the `create_person_value` method to separate the logic for creating a student and a teacher into their own methods within the `App` class.

```ruby
class App
  # ... (other methods)

  def create_student(age, name)
    # Logic to create a student
  end

  def create_teacher(age, name)
    # Logic to create a teacher
  end

  def create_a_person
    # ... (existing code)
    case is_student
    when 1
      create_student(age, person_name)
    when 2
      create_teacher(age, person_name)
    else
      puts 'Invalid option'
    end
    puts '🎉 Person added successfully.'
  end
end
```

- **YAGNI and KISS:**
  - Simplify the conditional checks in the `create_a_person` method using descriptive symbols or constants.

```ruby
class App
  STUDENT = 1
  TEACHER = 2

  def create_a_person
    # ... (existing code)
    case is_student
    when STUDENT
      create_student(age, person_name)
    when TEACHER
      create_teacher(age, person_name)
    else
      puts 'Invalid option'
    end
    puts '🎉 Person added successfully.'
  end
end
```

- **DRY (Don't Repeat Yourself):**
  - Extract the input gathering code for age and name into separate methods to avoid repetition.

```ruby
class App
  def input_age
    puts 'Age'
    gets.chomp.to_i
  end

  def input_name
    puts 'Name'
    gets.chomp
  end

  def create_a_person
    is_student = input_student_type
    age = input_age
    person_name = input_name

    case is_student
    when STUDENT
      create_student(age, person_name)
    when TEACHER
      create_teacher(age, person_name)
    else
      puts 'Invalid option'
    end
    puts '🎉 Person added successfully.'
  end
end
```

**Public Methods (create_a_book, create_a_rental, list_rentals):**
- **Single-Responsibility Principle:**
  - Move these methods inside the `App` class since they are closely related to its responsibilities.

```ruby
class App
  # ... (other methods)

  def create_a_book
    # ... (existing code)
  end

  def create_a_rental
    # ... (existing code)
  end

  def list_rentals
    # ... (existing code)
  end
end
```

Remember, these are just snippets to illustrate the suggested improvements. Make sure to integrate them into your existing code appropriately and ensure that everything works seamlessly together.